src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/.template.config/dotnetcli.host.json

```
@@ -1,6 +1,22 @@
 {
   "$schema": "https://json.schemastore.org/dotnetcli.host",
   "symbolInfo": {
+    "AiServiceProvider": {
+      "longName": "provider",
+      "shortName": "pr"
+    },
+    "VectorStore": {
+      "longName": "vector-store",
+      "shortName": "vs"
+    },
+    "UseManagedIdentity": {
+      "longName": "managed-identity",
+      "shortName": "mi"
+    },
+    "UseAspire": {
+      "longName": "aspire",
+      "shortName": "as"
+    },
```

Suggestions for single-character short names, but this is already much better. Cool that we can specify both long and short names like this, and using the kebab-cased long names is great!

```suggestion
    "AiServiceProvider": {
      "longName": "provider",
      "shortName": "p"
    },
    "VectorStore": {
      "longName": "vector-store",
      "shortName": "v"
    },
    "UseManagedIdentity": {
      "longName": "managed-identity",
      "shortName": "i"
    },
    "UseAspire": {
      "longName": "aspire",
      "shortName": "a"
    },
```

---

I had actually tried these, but the problem is that `v`, `i`, and `a` all appear to be reserved, so you have to escape them by doing `-p:v`, `-p:i`, etc. And then having another `p` short name on top of that makes it even a little more confusing 😄

```text
Template options:
  -F, --Framework <net9.0>                                     
  -p, --provider <azureopenai|githubmodels|ollama|openai>
  -p:v, --vector-store <azureaisearch|local|qdrant>
  -p:i, --managed-identity
  -p:a, --aspire
```

Just curious - what's the problem with multi-char short names? Is it that the widespread convention for CLI apps is to have single-char short names? I noticed that a lot of our templates currently use multi-char short names (and the universal options for `dotnet new` have `-lang` as the short option for `--language`).

Would we consider removing the short names completely and just having the long names? I saw that some of our other project templates do that (like `dotnet new console`).

---

> Would we consider removing the short names completely and just having the long names? I saw that some of our other project templates do that (like dotnet new console).

I personally don't expect many people to benefit from short names. Who creates projects often enough to benefit from that? Maybe us, but certainly not many people out there doing real jobs... :)

In some cases the CLI seems to auto-generate default short names for the parameters. Not sure if the `console` template explicitly suppresses that.

Altogether I'm fine with any choice here but do like the idea of not having short names so that we don't have to commit to any in particular and leave the options open.

---

> Who creates projects often enough to benefit from that? Maybe us, but certainly not many people out there doing real jobs... :)

Good point. I've gone ahead and explicitly removed the short names for now. We can always add them back later 🙂